packages/jest-mock/src/index.js

```diff
@@ -28,8 +28,16 @@ type MockFunctionResult = {
   /**
    * True if the function threw.
    * False if the function returned.
+   * Undefined if the call has not completed yet.
```

Comment:
This undefined value approach feels dirty, but I think it's the least intrusive solution for now. Other reasonable options are:
* Add another boolean flag to keep track of completions status, but then it still makes sense for "isThrow" to be undefined while "isComplete" is false, so then the information is duplicated.  Initalizing "isThrow" to false while "isComplete" is false seems misleading.
* Move forward with combining call arguments, invocation order, and result info into a single "MockCall" structure and get rid of the parallel arrays. Then the "result" property of the "MockCall" would be undefined until the call completes. Maybe we could soft-deprecate the existing "calls" and "invocationOrder" arrays by keeping them for now and duplicating the info in the new structure. What would we call the array of these all-encompassing "MockCall" objects? An obvious name is "calls", but that's what the current array of call args is currently named already.

Reply:
I think combining them into a single data structure would be best in the long run. But to avoid breaking the entire world, I think the `undefined` approach is good enough for now.

Reply:
An alternative is to make a parallel `mockStatus: 'returned' | 'threw' | 'incomplete'` string (enum) instead of a boolean

Reply:
Are you OK with making a breaking change of removing `isThrow` and replacing it with a string enum property?

Looking to the future (eventually when it's OK to break the entire world) when `MockFunctionResult` will only be used to represent completed calls, I think something like `type: 'return' | 'throw';` would be more intuitive. And for now, add in the `'incomplete'` value.

Reply:
Yes, I think it's safe to say no one is using this yet as long as the matchers work (though I'm biased since that's the API recommended on the original api 😄)

Reply:
I caved in to peer pressure and switched to a string enum :)

I think I resisted so much because I spent so many years working with plain, untyped, Javascript, where it was hard to discover what magic string values were valid for such a field, and there were no tools to catch typos/mistakes.